core/shared/platform/linux/platform_internal.h

```diff
@@ -122,6 +122,10 @@ os_sigreturn();
 void
 os_set_signal_number_for_blocking_op(int signo);
 
+typedef int os_handle;
+typedef DIR os_dir_stream;
```

Comment:
In `posix.c`, we store the directory handle via a pointer and I'd prefer to keep it that way. It makes it simpler to check if it's been initialised or not. Also, I doubt any platform could define this as a fundamental type so we would pretty much always want to keep a pointer to it and not store the struct/object directly in the `fd_object` type.

Reply:
It still should be ok since we're not relying on the internal implementation of 'DIR'. We only pass it to the dirent functions in the posix file implementation.

Reply:
you are assuming a directory handle is a pointer to something. it might not be true for other platforms.

Reply:
I think that's a reasonable assumption but will change it to be a non-pointer type. It will require adding another function like `os_get_invalid_dir_stream`.